lib/index.js

```diff
@@ -2108,6 +2109,181 @@ addMethod('toPromise', function (PromiseCtor) {
+    if (isObserver) {
+        return onNext;
+    }
+
+    // Otherwise create an observer object
+    return {
+        next: onNext,
+        error: onError,
+        complete: onComplete,
+    };
+}
+
+/**
+ * Consumes values using the Observable subscribe signature. Unlike other
+ * consumption methods, subscribe can be called multiple times. Each
+ * subscription will receive the current value before receiving the next value.
+ * Subscribing to an already consumed stream will result in an error.
+ *
+ * Implements the Observable subscribe functionality as defined by the spec:
+ * https://tc39.github.io/proposal-observable/#observable-prototype-subscribe
+ *
+ * @id subscribe
+ * @section Consumption
+ * @name Stream.subscribe(onNext, onError, onComplete)
+ * @param {Function|object|null} onNext - Handler for next value or observer
+ * @param {Function|null} onError - Handler function for errors.
+ * @param {Function|null} onCompleted - Handler Function when stream is done.
+ * @api public
+ *
+ * // with callbacks
+ * _([1, 2, 3, 4]).subscribe(
+ *     function onNext (x) {
+ *         // Called for each value that comes downstream
+ *         console.log('Received onNext value', x);
+ *     },
+ *     function onError (err) {
+ *         // Called one time with error or zero if no errors occur upstream
+ *         console.error('Single highland stream error', err);
+ *     },
+ *     function onComplete () {
+ *         // Receives no arguments
+ *         // Called only once when stream is completed.
+ *         console.log('Completed!');
+ *     }
+ * );
+ *
+ * // with an observer
+ * _([1, 2, 3, 4]).subscribe({
+ *     next (x) {
+ *         console.log('Received next value', x);
+ *     },
+ *     error (err) {
+ *         console.error('An error occurred upstream', err);
+ *     },
+ *     complete () {
+ *         console.log('Completed!')
+ *     }
+ * });
+ */
+
+addMethod('subscribe', function (onNext, onError, onComplete) {
+    var self = this;
+    var observer = createObserver(onNext, onError, onComplete);
+
+    // Don't let users subscribe to an already completed stream
+    if (self.ended && observer.error) {
+        observer.error(new Error('Subscribe called on an already completed stream.'));
+    }
+
+    var subscription = ObservableSubscription.create();
+    var source = self.fork().consume(function (err, x, push, next) {
+        if (err) {
+            push(null, nil);
+
+            if (observer.error) {
+                observer.error(err);
+            }
+            ObservableSubscription.cleanup(subscription);
+        }
+        else if (x === nil) {
+            if (observer.complete) {
+                observer.complete();
+            }
+            ObservableSubscription.cleanup(subscription);
+        }
+        else {
+            if (observer.next) {
+                observer.next(x);
+            }
+            next();
+        }
+    });
+    subscription.source = source;
```

Comment:
To avoid the circular dependency, you can destroy the `fork` rather than the `consume` stream.

`consume` can emit data synchronously, so it's not necessarily safe to operate on an incomplete `ObservableSubscription`. Like this:

```javascript
var source = self.fork();
var subscription = new ObservableSubscription(source);
source.consume(...).resume();
```